src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -194,6 +197,38 @@ public void GetItems<T>(ReadOnlySpan<T> choices, Span<T> destination)
                 throw new ArgumentException(SR.Arg_EmptySpan, nameof(choices));
             }
 
+            // The most expensive part of this operation is the call to get random data. We can
+            // do so potentially many fewer times if:
+            // - the number of choices is <= 256. This let's us get a single byte per choice.
+            // - the number of choices is a power of two. This let's us use a byte and simply mask off
+            //   unnecessary bits cheaply rather than needing to use rejection sampling.
+            // In such a case, we can grab a bunch of random bytes in one call.
+            if (BitOperations.IsPow2(choices.Length) && choices.Length <= 256)
+            {
+                Span<byte> randomBytes = stackalloc byte[512]; // arbitrary size, a balance between stack consumed and number of random calls required
```

Comment:
The line length with this comment post-statement feels a bit long.

```suggestion
                // This is an arbitrary size, a balance between stack consumed and number of random calls required
                Span<byte> randomBytes = stackalloc byte[512];
```

looks happier (more in line with the rest of the file (or, at least, PR))